### PR TITLE
Added tests for convertHsdsType

### PR DESCRIPTION
**Pull request by loichuder**
Follow up of #452

### CODE REVIEW

File: src/h5web/providers/hsds/models.ts [hunk 1]
```diff
@@ -21,15 +21,16 @@ export type HsdsType =
   | HsdsBaseType
   | HsdsArrayType
   | HsdsVLenType
-  | HsdsCompoundType;
+  | HsdsCompoundType
+  | HDF5Id;
```
Comment (loichuder) [inline src/h5web/providers/hsds/models.ts:25]: Oops, forgot in the interface that the type can be `HDF5Id`

File: src/h5web/providers/hsds/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          },
+        ],
+      });
+    });
+
+    it('should throw when encountering an unknown type', () => {
+      const unknownType = { class: 'NO_CLASS' };
+      expect(() => convertHsdsType(unknownType as HsdsType)).toThrow(
+        /Unknown type/u
+      );
```
Comment (loichuder) [inline src/h5web/providers/hsds/utils.test.ts:176]: Even if the error cannot happen according to TypeScript, we need to test it as TS has no control on what comes out of the HSDS requests.

File: src/h5web/providers/hsds/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        base: {
+          class: HDF5TypeClass.Integer,
+          size: 8,
+          endianness: 'LE',
+          unsigned: undefined,
+        },
+      });
+    });
+
+    it('should convert the field types of Compound type', () => {
```
Comment (loichuder) [inline src/h5web/providers/hsds/utils.test.ts:118]: The test might seem over-complicated but it makes sure that all types are converted (even deeply nested ones).

It cannot happen in `Array` in `VLen` types as the base is a `HsdsBaseType` in these cases.

File: src/h5web/providers/hsds/utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        class: HDF5TypeClass.VLen,
+        base: integerType,
+      };
+      const compoundType: HsdsCompoundType = {
+        class: HDF5TypeClass.Compound,
+        fields: [
+          {
+            name: 'f1',
+            type: floatType,
+          },
```
Comment (axelboc) [inline src/h5web/providers/hsds/utils.test.ts:132]: maybe shrink those onto single lines?

File: src/h5web/providers/hsds/utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+          {
+            name: 'f2',
+            type: {
+              class: HDF5TypeClass.VLen,
+              base: {
+                class: HDF5TypeClass.Integer,
+                size: 8,
+                endianness: 'LE',
+              },
```
Comment (axelboc) [inline src/h5web/providers/hsds/utils.test.ts:158]: You could probably reuse some of the expected `base` types from previous tests by storing them in constants at the top, like we do in other test files.

File: src/h5web/providers/hsds/utils.ts [hunk 6]
```diff
@@ -52,12 +52,19 @@ export function convertHsdsBaseType(hsdsBaseType: HsdsBaseType): HDF5BaseType {
...
-    unsigned: sign === 'U' || undefined,
   };
+  if (sign === 'U') {
+    return {
+      ...hdf5Type,
+      unsigned: true,
+    };
+  }
+
+  return hdf5Type;
```
Comment (axelboc) [inline src/h5web/providers/hsds/utils.ts:67]: Always annoying this one...

```ts
return {
  // ...
  ...(sign === 'U' ? { unsigned: true } : {})
};
```

**Review by axelboc — APPROVED** [review]

File: src/h5web/providers/hsds/utils.ts [hunk 7]
```diff
@@ -52,11 +52,15 @@ export function convertHsdsBaseType(hsdsBaseType: HsdsBaseType): HDF5BaseType {
...
     class: hsdsClass,
     endianness: endianness as HDF5Endianness,
     size: Number.parseInt(size, 10),
-    unsigned: sign === 'U' || undefined,
+  };
+
+  return {
+    ...hdf5Type,
+    ...(sign === 'U' ? { unsigned: true } : {}),
   };
```
Comment (axelboc) [inline src/h5web/providers/hsds/utils.ts:64]: oh, no need for the separate return here 😉

**Review by axelboc — APPROVED** [review]

File: src/h5web/providers/hsds/utils.ts (hunk 7 above)
Comment (loichuder) [reply src/h5web/providers/hsds/utils.ts:64]: :facepalm: thanks

**Pull request merged**